### ISSUE TITLE
Allow GcpProjectIdProvider to provide the numerical value for the project

### ISSUE THREAD

**Issue by Alos** [spring-cloud/spring-cloud-gcp#2046]
GcpProjectIdProvider currently only provides the string value that contains the name of the project, it would be useful to have it also contain the numerical value associated with the project.

**Comment by meltsufin** [comment]
I think in that in terms of the single responsibility principle, we should keep `GcpProjectIdProvider` simple. However, we might want to provide a template for the Project Metadata Service. How common is the need for the project number?

**Comment by stephen-harlow** [comment]
I'd say that it's somewhat common, but not as common as the `project-id`. For example, if you want to know what the name of the service account is for a specific GCP project (The one that is automatically created for Builds and Compute Engine.), the only way to currently do that is through something like: [this](https://stackoverflow.com/questions/52307471/get-google-cloud-project-number-using-its-id).

**Comment by meltsufin** [comment]
Sounds like auto-configuration for the Resource Manager API or even a template for it can be useful in case like this one.

**Comment by stephen-harlow** [comment]
Yeah, having a way to ask the Resource Manager API directly `What is the Project Number?` would be useful.

**Comment by elefeint** [comment]
Note that Resource Manager API is disabled by default, so if we provide an autowired object, we should at minimum document it as the most common issue.

There is an example of [acquiring ResourceManager](https://github.com/spring-cloud/spring-cloud-gcp/blob/28e4aeb61c7da2b2a11f46d0df396eeb527bec57/spring-cloud-gcp-security-iap/src/main/java/org/springframework/cloud/gcp/security/iap/AppEngineAudienceProvider.java#L40) client library abstraction and [getting a project number](https://github.com/spring-cloud/spring-cloud-gcp/blob/28e4aeb61c7da2b2a11f46d0df396eeb527bec57/spring-cloud-gcp-security-iap/src/main/java/org/springframework/cloud/gcp/security/iap/AppEngineAudienceProvider.java#L58) from it in IAP.

**Comment by meltsufin** [comment]
I think most APIs are disabled by default.
